### DISCUSSION TITLE
Cockpit plugin or WebUI

### DISCUSSION THREAD

**Discussion by ghost** [ctc-oss/fapolicy-analyzer#902] [Ideas]

**Comment by jw3** [comment]
Thanks @ryancoeit, glad to hear it has been useful to you.

Could you elaborate a little more on the web ui use case and on why fapolicyd would not need to be put in debug mode to identify what is being blocked?

Thanks!

**Reply by jw3** [reply]
> im trying to avoid is having to stop fapolicyd in order to then run it again in a debug mode

This sounds like the [policy profiler tool](https://github.com/ctc-oss/fapolicy-analyzer/wiki/User-Guide#policy-profiler).

> view the logs in a live format, scroll backwards etc to find the culprit and add rules with a systemd reload so that we dont have to drop the shield so to speak

This sounds like the [policy analyzer tool](https://github.com/ctc-oss/fapolicy-analyzer/wiki/User-Guide#policy-analyzer).

Perhaps the documentation is not clear enough about the ability to use the policy analyzer without the policy profiler.  [This section](https://github.com/ctc-oss/fapolicy-analyzer/wiki/User-Guide#supported-event-log-sources) describes the two debug mode alternatives.

Does that functionality that currently exists address your concerns?

> It would be great if there were a cockpit module that would allow you to view the logs in a live format

Sure a web tool for this would be pretty cool. This team does not currently plan on anything like this, but if you or someone else was interested in using the crates from this project, we would publish them to crates.io.  How compatible the crates would be in a web framework is an unknown, but we would consider pull requests that bring things into alignment.

Thanks

**Reply by jw3** [reply]
Maybe the docs need some text along the lines of 

> The Policy Analysis tool can access log data in one of two ways: (1) Read from a supported log source (debug file, syslog, audit log) or (2) Load the log captured at the end of a Policy Profiler run. Reading the logs does not require any changes to fapolicyd to analyze, avoiding the need to take fapolicyd offline and run in debug mode, as is required by the Policy Profiler.
